### ISSUE TITLE
Newest 0.2.5 version throws Login error...

### ISSUE THREAD

**Issue by MDashK** [SwipeX/PokeMate#185]
Using default config.properties location
com.pokegoapi.exceptions.LoginFailedException: svarzee.gps.gpsoauth.Gpsoauth$Tok
enRequestFailed
        at com.pokegoapi.auth.GoogleAutoCredentialProvider.login(GoogleAutoCrede
ntialProvider.java:53)
        at com.pokegoapi.auth.GoogleAutoCredentialProvider.<init>(GoogleAutoCred
entialProvider.java:41)
        at dekk.pw.pokemate.Context.Login(Context.java:70)
        at dekk.pw.pokemate.Context.Login(Context.java:39)
        at dekk.pw.pokemate.PokeMate.<init>(PokeMate.java:59)
        at dekk.pw.pokemate.PokeMate.main(PokeMate.java:93)
Caused by: svarzee.gps.gpsoauth.Gpsoauth$TokenRequestFailed
        at svarzee.gps.gpsoauth.Gpsoauth.performMasterLoginForToken(Gpsoauth.jav
a:105)
        at svarzee.gps.gpsoauth.Gpsoauth.performMasterLoginForToken(Gpsoauth.jav
a:88)
        at com.pokegoapi.auth.GoogleAutoCredentialProvider.login(GoogleAutoCrede
ntialProvider.java:46)
        ... 5 more
Logged in as *****************@gmail.com
Exception in thread "main" com.pokegoapi.exceptions.LoginFailedException: Creden
tial Provider is null
        at com.pokegoapi.api.PokemonGo.<init>(PokemonGo.java:75)
        at dekk.pw.pokemate.PokeMate.<init>(PokeMate.java:62)
        at dekk.pw.pokemate.PokeMate.main(PokeMate.java:93)
Press any key to continue . . .

**Comment by rherartt** [comment]
I am having an error with the new version too:

at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)

**Comment by Messna** [comment]
Can you login using the app? I think the servers are just down again: http://cmmcd.com/PokemonGo/

**Comment by rherartt** [comment]
oh probably that then, sorry!

**Comment by MDashK** [comment]
I am logged in with ver. 0.2.1. No issues. Still tagging PokeStops and everything.

**Comment by rherartt** [comment]
server on and I still have this:

at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)
        at java.util.regex.Pattern$LazyLoop.match(Unknown Source)
        at java.util.regex.Pattern$GroupTail.match(Unknown Source)
        at java.util.regex.Pattern$BranchConn.match(Unknown Source)
        at java.util.regex.Pattern$CharProperty.match(Unknown Source)
        at java.util.regex.Pattern$Branch.match(Unknown Source)
        at java.util.regex.Pattern$GroupHead.match(Unknown Source)

**Comment by Dragon92** [comment]
@rherartt I second that, same exception.

**Comment by Darius2006** [comment]
I have the same problem with the new version...

**Comment by coolgigahub** [comment]
I got excatly the same error.

**Comment by SippieCup** [comment]
Please make a new issue for the regex error, this thread is for a login authentication error and will not get as much attention as it has been solved. 

Off topic conversations should take place in their own threads.

**Comment by MDashK** [comment]
The thread/issue as NOT been solved...
I'm the original OP and I'm still having the issue...

**Comment by SippieCup** [comment]
The issue you have is that the Google login servers are down. So it won't work until they come back up. 

There's nothing that needs to be done other than watching ispokemongodownornot.com

**Comment by kylestev** [comment]
waiting on this to be merged to resolve the stack overflow. svarzee/gpsoauth-java#4

**Comment by MDashK** [comment]
@SippieCup: Google servers down? That's weird, because It's working on the previous version v0.2.1, as I stated earlier...
Only have this error in this new version.

The error the user "rherartt" reported is NOT THE SAME I'm having and opened this issue relating to.

**Comment by vmarchaud** [comment]
Maintainers of Java API here, you should switch over `GoogleUserCredentialProvider` that has been pushed to our dev branch few hours ago, it's almost the same as the old one, both based on tokens over plain password.

**Comment by kylestev** [comment]
@vmarchaud thanks for letting us know! someone is working on a fix

**Comment by SippieCup** [comment]
Fixed in my new release, Compiling from source might be a problem until jitpack resolves its issues.

https://github.com/SippieCup/PokeMate/releases/tag/2.3.5

**Comment by veskoz** [comment]
Tried with:
- pokemate-SippieCup-0.2.3.5.jar  
- pokemate-SwipeX-0.2.3.5.jar
- PokeMate-0.2.5.jar

Still got this:
com.pokegoapi.exceptions.LoginFailedException: svarzee.gps.gpsoauth.Gpsoauth$TokenRequestFailed
        at com.pokegoapi.auth.GoogleAutoCredentialProvider.login(GoogleAutoCredentialProvider.java:53)
        at com.pokegoapi.auth.GoogleAutoCredentialProvider.<init>(GoogleAutoCredentialProvider.java:41)
        at dekk.pw.pokemate.Context.Login(Context.java:73)
        at dekk.pw.pokemate.Context.Login(Context.java:39)
        at dekk.pw.pokemate.PokeMate.<init>(PokeMate.java:59)
        at dekk.pw.pokemate.PokeMate.main(PokeMate.java:93)
Caused by: svarzee.gps.gpsoauth.Gpsoauth$TokenRequestFailed
        at svarzee.gps.gpsoauth.Gpsoauth.performMasterLoginForToken(Gpsoauth.java:105)
        at svarzee.gps.gpsoauth.Gpsoauth.performMasterLoginForToken(Gpsoauth.java:88)
        at com.pokegoapi.auth.GoogleAutoCredentialProvider.login(GoogleAutoCredentialProvider.java:46)
        ... 5 more
Logged in as  //here goes my account, deleted for security
Exception in thread "main" com.pokegoapi.exceptions.LoginFailedException: Credential Provider is null
        at com.pokegoapi.api.PokemonGo.<init>(PokemonGo.java:75)
        at dekk.pw.pokemate.PokeMate.<init>(PokeMate.java:62)
        at dekk.pw.pokemate.PokeMate.main(PokeMate.java:93)

**Comment by MDashK** [comment]
With version "pokemate-SippieCup-0.2.3.5", I can login without issue.
BUT, if I delete the "tokens" folder, along with the TXT file inside of it, and try to run it, it will spit the initial error I reported in the first comment.
Other than that, seems to be running OK.

Still no luck with "PokeMate-0.2.5" version.

**Comment by veskoz** [comment]
Maybe you can because you don't have the google authentication, the thing that asks you to provide two numbers to a given url typically www.google.com/device
Or because you have alredy done this and it's saved in the token folder.
I GUESS that if you just create the folder tokens it works.
Still no luck for me, FeelsBadMan.

**Comment by Dragon92** [comment]
OK, I managed to suss out how to get around the regex errors: instead of using "XXXXX@gmail.com" for the username, use just "XXXXX". That seems to take care of that issue, however the errors @veskoz pointed out still prevent the bot from running.

**Comment by SwipeX** [comment]
I'm working on a fix atm

**Comment by SippieCup** [comment]
Has this been resolved?

**Comment by veskoz** [comment]
With 3.0 i Can login.

**Comment by SippieCup** [comment]
Resolved. Closing Thread.
